src-cpp/rdkafkacpp_int.h

```diff
@@ -391,32 +391,144 @@ class ConfImpl : public Conf {
   }
 
   Conf::ConfResult get(const std::string &name, std::string &value) const {
+	  if (name.compare("dr_cb") == 0 ||
+	      name.compare("event_cb") == 0 ||
+	      name.compare("partitioner_cb") == 0 ||
+	      name.compare("partitioner_key_pointer_cb") == 0 ||
+	      name.compare("socket_cb") == 0 ||
+	      name.compare("open_cb") == 0 ||
+	      name.compare("rebalance_cb") == 0 ||
+	      name.compare("offset_commit_cb") == 0 ) {
+	      return Conf::CONF_INVALID;
+	  }
 	  size_t size;
 	  rd_kafka_conf_res_t res = RD_KAFKA_CONF_OK;
 	  if (rk_conf_) {
 		  if ((res = rd_kafka_conf_get(rk_conf_,
 			  name.c_str(), NULL, &size)) != RD_KAFKA_CONF_OK)
 			  return static_cast<Conf::ConfResult>(res);
 
-		  value.resize(size);
-		  if ((res = rd_kafka_conf_get(rk_conf_, name.c_str(),
-			  (char *)value.c_str(), &size)) != RD_KAFKA_CONF_OK)
-			  return static_cast<Conf::ConfResult>(res);
+		  char valueCharArray[1000];
+		  res = rd_kafka_conf_get(rk_conf_, name.c_str(),
+				      (char *) valueCharArray, &size);
+		  value.assign(valueCharArray);
```

Comment:
There is no point in querying for the conf_get size above if we're not going to use it.
And what if the value is larger than this fixed size?
Please use the returned size